enbaskakova-DOCSUP-652 (#101)
BayoNet/ClickHouse: docs/en/sql_reference/aggregate_functions/combinators.md

## -OrDefault {#agg-functions-combinator-ordefault}

> This combinator replaces returned value with default one, if the aggregate function does not have values to calculate. Works with different aggregate functions.

Reviewer: ```suggestion
Changes behavour of an aggregate function.

If an aggregate function doesn't have input values, with this combinator it returns the default value for its return data type. Applies to the aggregate functions that can take empty input data.
```

`-OrDefault` can be combined with other combinators.

**Syntax** 

``` sql
<aggFunction>OrDefault(x)
```

**Parameters**

> - `x` - Parameter of the aggregate function. A type of the parameter depends on the specific aggregate function.

Reviewer: ```suggestion
- `x` — Aggregate function parameters.
```

**Returned values** 
 

> Returns default value of the aggregate function’s return type if there is nothing to aggregate.

Reviewer: ```suggestion
Returns the default value of an aggregate function’s return type if there is nothing to aggregate.
```

> Type depends on the specific aggregate function.

Reviewer: ```suggestion
Type depends on the aggregate function used.
```

**Example**

Query:

``` sql
SELECT avg(number), avgOrDefault(number) FROM numbers(0)
```

Result:

``` text
┌─avg(number)─┬─avgOrDefault(number)─┐
│         nan │                    0 │
└─────────────┴──────────────────────┘
```

Also `-OrDefault` can be used with a few combinators. It is useful when the aggregate function does not accept the empty input.

Query:

``` sql
SELECT avgOrDefaultIf(x, x > 10)
FROM
(
    SELECT toDecimal32(1.23, 2) AS x
)
```

Result:

``` text
┌─avgOrDefaultIf(x, greater(x, 10))─┐
│                              0.00 │
└───────────────────────────────────┘
```

**See also**

> - [Functions For Working With External Dictionaries](../functions/ext_dict_functions.md#ext_dict_functions-other) - `-OrDefault` is used with functions for working with external dictionaries.

Reviewer: ```suggestion
- [Functions For Working With External Dictionaries](../functions/ext_dict_functions.md#ext_dict_functions-other) — `-OrDefault` is used with functions for working with external dictionaries.
```

Reviewer: Does the function dictGet and others use OrDefault as a combinator?  I found dictGetOrDefault as a separate function. I think we should delete this link.

> ## -OrNull {#agg-functions-combinator-ornull}

Reviewer: Apply my comments to this function too.

> This combinator replaces returned value with nullable one, if the aggregate function does not have values to calculate. Works with different aggregate functions. 

Reviewer: Nullable is a data type, but we talk about the specific value NULL. The function can return NULL in case of empty input data set and/or it can convert the return data type in the Nullable data type.

Check it please and update the description.
